### DISCUSSION TITLE
HERO13 1.30.70 out today (2024 Nov 21.)

### DISCUSSION THREAD

**Discussion by dnewman-gpsw** [gopro/labs#1179] [Announcements]
As it is only been a few weeks, this is not a feature packed update.  However it does have some interesting extensions.

- Added **WBDV** - White Balance DiVe improvements. Rather than WARM for improving diving white balance, which effects WB the same at all depths, WBDV is more automatic – as the scene get more blue, the more the red channel is gain up. Currently ```$WBDV=1``` is uses =2.33,4.0 internally (gaining red 4x after red gain hits 2.33), so you can try tuning this yourself e.g. ```$WBDV=2.4,5.0```
- Added **SHEL** - Using ```$SHEL=1``` enables a command shell for Labs over a USB serial port. Every experimental, and looking for feedback. Connect to COM at 115200 8N1. All QR code command work, just type them. Used $ commands will report their data back to the shell, e.g. $BITR or $A or $p:T to read variables.
- Added **OPNW** - Use ```$OPNW=1``` to enable an Open Network (allowing faster non-secure http:// vs https:// to be used with Open GoPro)

Always looking for feedback. 

Here is a video demo of the new shell: https://www.youtube.com/watch?v=88V2EaM4ahU

**Comment by nandodiver** [comment]
FW 01.30.70 
Small bug - not possible to rename the preset/profile name in Quick Pro, it freezes the app

Looking forward to test the WBDV diving this weekend 

<img width="925" alt="Captura de ecrã 2024-12-07, às 19 51 20" src="https://github.com/user-attachments/assets/4bf906f3-1f68-4b94-835c-2bf1c6ef6866">

**Reply by dnewman-gpsw** [reply]
Quik doesn't have any direct support for Labs. So that can happen.

A bug has been found in WBDV command causing the red to go to zero (an overflow.) It will be fixed soon.

**Comment by nandodiver** [comment]
Upgrade to v1.30.72, December 13, 2024

I had to !RESET for some reason the WBLK got permanent to all presets

Ive Check with metadata and If i scan the code $WBDV=1 and add name WBDV it get permanent in all presets, if i dont add a name is only visible in the select preset.

I add the code $WBDV=2.33, 4.0 and it appears in the metadata visible has DIVE=1 only in the select scan preset.

Its confusing because $DIVE=1 is the water refrex index (RI) 

It was supposed to show WBDV=1 in the metadata

<img width="843" alt="Captura de ecrã 2024-12-16, às 00 46 33" src="https://github.com/user-attachments/assets/362ebd0e-cb88-47fc-85f2-7b7b58a28e26" />

**Reply by dnewman-gpsw** [reply]
If you scan $WBDV=1, you wouldn't need it is a named preset, and the $WBDV is global (effects all preset.)  Only name a preset WBDV will isolate the extension to that preset.

$WBDV=2.33,4.0 (don't use spaces) is the same as $WBDV=1

>Its confusing because $DIVE=1 is the water refrex index (RI)

$DIVE=1 is the same as $DIVE=1.335, it is just saving people to remember the RI of water. All extension are activated with 1, so DIVE=1, couldn't be the RI of air, and that would be deactivated state.  So there is a consistency and logic to this.

Where are ATIO, SO80 and  ISO8 coming from?  They aren't Labs extensions.

**Reply by nandodiver** [reply]
> If you scan $WBDV=1, you wouldn't need it is a named preset, and the $WBDV is global (effects all preset.) Only name a preset WBDV will isolate the extension to that preset.
> 
> $WBDV=2.33,4.0 (don't use spaces) is the same as $WBDV=1
> 
> > Its confusing because $DIVE=1 is the water refrex index (RI)
> 
> $DIVE=1 is the same as $DIVE=1.335, it is just saving people to remember the RI of water. All extension are activated with 1, so DIVE=1, couldn't be the RI of air, and that would be deactivated state. So there is a consistency and logic to this.
> 
> Where are ATIO, SO80 and ISO8 coming from? They aren't Labs extensions.

ATIO, SO80, IS08, i have no idea 
to be clear, If i scan $WBDV=1 the metadata dont show WBDV=1 but shows DIVE=1 
On my second post using previous version WBDV=1 was showing in metadata 

I will reset and reinstall FW a again.

**Comment by nandodiver** [comment]
it looks like the !Reset is not resetting 

<img width="1017" alt="Captura de ecrã 2024-12-16, às 11 02 01" src="https://github.com/user-attachments/assets/81f7291d-ec72-4e41-b7c9-20ebf558fcc8" />
<img width="966" alt="Captura de ecrã 2024-12-16, às 11 02 32" src="https://github.com/user-attachments/assets/381fe27e-4296-4b6b-a236-d270fb4a8dde" />
<img width="998" alt="Captura de ecrã 2024-12-16, às 11 03 09" src="https://github.com/user-attachments/assets/78e5a6a3-1968-49b0-a2ab-04f97d1857da" />

**Reply by dnewman-gpsw** [reply]
!Reset is not a command, try !RESET with the required button presses.

**Comment by nandodiver** [comment]
re-install stock firmware H24.01.01.30.00
have to done factory settings to remove all the presets from gopro labs

metadata now showing this 
<img width="950" alt="Captura de ecrã 2024-12-16, às 11 18 48" src="https://github.com/user-attachments/assets/195c753a-12c1-4b7f-a554-ebca372837c9" />

**Comment by nandodiver** [comment]
reinstall gopro labs firmware v1.30.72, December 13, 2024

metadata looks normal 

<img width="1007" alt="Captura de ecrã 2024-12-16, às 11 24 48" src="https://github.com/user-attachments/assets/28504e65-9aa4-4b48-a69e-d10b75dce15d" />

**Comment by nandodiver** [comment]
I have scanned new codes

Preset - $WBDV=1
preset - *BITR=150  
preset - $WBLK=1

Looks like the metadata now working fine 

<img width="889" alt="Captura de ecrã 2024-12-16, às 16 38 01" src="https://github.com/user-attachments/assets/5c2be2c8-df14-4470-80ca-d277f6c92d20" />
<img width="874" alt="Captura de ecrã 2024-12-16, às 16 38 44" src="https://github.com/user-attachments/assets/38773540-aebe-4037-86e2-3f65e436fca2" />
<img width="1019" alt="Captura de ecrã 2024-12-16, às 16 39 08" src="https://github.com/user-attachments/assets/f4f46933-e5e9-4df3-8f8f-da61b68ffe38" />

**Reply by dnewman-gpsw** [reply]
You still are adding random stuff: O160, TOWB, 1600 are added.  Can you send me your qrlog.txt file (from the MISC folder.)

**Reply by nandodiver** [reply]
sure 
[qrlog.txt](https://github.com/user-attachments/files/18154004/qrlog.txt)

**Reply by dnewman-gpsw** [reply]
Also you don't need to screen grab it all.  You can copy and paste the user metadata section, just remove the ```<html><body><!--StartFragment-->```  and ```<!--EndFragment--></body></html>``` and it comed out pretty (from one of my cameras):

DEVC |  
-- | --
DVID | USRM
DVNM | User Metadata
STRM |  
DLTM |  
24HZ | 1
RLTC | 0
SYID | 97410
TUSB | 1
DBGL | 1
EXPS | 0
NR01 | 1
DRFT | 1
DRFL | 153753760, 604800
DRFS | -9.6512
SYTM | 241113132238.956
USRD | 1

**Reply by dnewman-gpsw** [reply]
I will need qrlog.bak (also) as the other is not showing what you scanned.  You custom presets are messed up, so you are scanning incorrect QR Codes, or there is a bug.  

This is wrong, at suggests you have a preset named "$TOWB=1$1600=1"  Why?
```[2024-12-16 16:14:35 63%   ] preset CMD:!N$TOWB=1$1600=1```

**Reply by nandodiver** [reply]
i cant find qrlog.bak

i only have qr.bin

**Comment by nandodiver** [comment]
I think the metadata TOWB - 0160 -1600 are related with personalised name of the preset 
![IMG_5850](https://github.com/user-attachments/assets/fcf67f41-b3a5-4915-963c-a09a7c899a7c)

**Reply by dnewman-gpsw** [reply]
That is not technically not supported.  If you using WBDV in the preset name, you should only use other valid Fourccs, otherwise the commands could be rejected.  There is a bug, as these should be all rejected and Labs names, the code is being too accepting.

**Reply by nandodiver** [reply]
ok, im reinstalling all and try different preset name, to not interfering with metadata.

**Reply by dnewman-gpsw** [reply]
reinstall is not neccessary, or helpful.  Just use !PRESET (to reset the presets) and !RESET (to clear any stale Labs settings)

**Reply by nandodiver** [reply]
There a bug in quick app that prevent to change the name.

**Reply by dnewman-gpsw** [reply]
Don't you use Quik for this, just reset the presets.

**Reply by nandodiver** [reply]
whats the code for White balance lock ? 

$WBLK=1 
or the one from QRcode app oMBLK=1
![IMG_5851](https://github.com/user-attachments/assets/81bf3ac0-6dec-4a6e-9345-a9c158d0e9a0)

**Reply by dnewman-gpsw** [reply]
Both. Go to the about tabs, and select 2024 codes.
![image](https://github.com/user-attachments/assets/5e533261-4ef5-4ca8-a926-8df6e7c7f23c)

**Comment by nandodiver** [comment]
I think there a bug in version v1.30.72, December 13, 2024 

I just did a !RESET, install stock firmware, reinstall pro labs FW 

Kept Video preset and create 2 more video preset with random name BRANCO BLOCK and RED4X DIVE 

**Haven't scanned any code yet** 

Made 3 videos with one of each preset,  here's the metadata, it show  DIVE and ED4X (this im sure are the preset name )
<img width="257" alt="Captura de ecrã 2024-12-16, às 19 52 24" src="https://github.com/user-attachments/assets/2f0f6c0a-0c8a-4f03-be4f-0e8d295846bf" />

<img width="285" alt="Captura de ecrã 2024-12-16, às 19 53 15" src="https://github.com/user-attachments/assets/11b9a28b-9cd3-4748-b357-36c92ae7b51c" />
<img width="218" alt="Captura de ecrã 2024-12-16, às 19 54 41" src="https://github.com/user-attachments/assets/c6a01a94-a31d-457d-8bcb-90a18835b2db" />

![IMG_5852](https://github.com/user-attachments/assets/d69062ee-9fd1-4249-ad73-46bf403b355d)

**Reply by dnewman-gpsw** [reply]
Yes that is a bug. Fortunately is it easily avoided using mixed case: use "Branco Block" and "Red4x Dive".  Then there is no confusion.

**Reply by nandodiver** [reply]
Ive reset the custom name presets and use pre-defined presets


Video 
<img width="211" alt="Captura de ecrã 2024-12-16, às 20 18 58" src="https://github.com/user-attachments/assets/d32f4d10-306f-4411-9db5-8341a0c3b726" />

bike 

<img width="258" alt="Captura de ecrã 2024-12-16, às 20 19 24" src="https://github.com/user-attachments/assets/b6da88b3-54f4-43ba-8b36-80481b3f4ddf" />

air 

<img width="246" alt="Captura de ecrã 2024-12-16, às 20 19 53" src="https://github.com/user-attachments/assets/31f9e433-7067-4a06-b1a0-e169d6f488a9" />

**Reply by dnewman-gpsw** [reply]
Mixed-case names should work fine.

**Reply by nandodiver** [reply]
Ive got back to custom name presets and scanned the qrcodes (something not right) 

Fist one to scan was *BITR=150 and made a bip sound when scanned in **Video** 
Second to scan was  $WBLK=1  and no sound was made when scanning in **Branco Blok**
Third to scan was $WBDV=1 and no sound was made when scanning in **Red4x**

Then made 3 videos, here's the metadata 

video  *BITR=150

<img width="311" alt="Captura de ecrã 2024-12-16, às 21 14 54" src="https://github.com/user-attachments/assets/393dc5df-5aac-4b9a-a298-1d7bfbd8b384" />

Red4x $WBDV=1 

<img width="234" alt="Captura de ecrã 2024-12-16, às 21 19 54" src="https://github.com/user-attachments/assets/131a501c-fce7-4849-a4b1-544517bd2182" />

Branco Blok $WBLK=1 

<img width="215" alt="Captura de ecrã 2024-12-16, às 21 22 21" src="https://github.com/user-attachments/assets/7893ea2d-7007-4d53-acd2-0b4cb94fb0fd" />

**Reply by dnewman-gpsw** [reply]
I don't see the issue.  Beeps are not required.

**Reply by nandodiver** [reply]
!RESET .... scanning just the $WBLK=1 in one preset the presets will have WBLK has well 
all metadata are the same 

the $ is acting like if was * 

in previous version i could have 3 diferent presets with 3 diferents codes 

<img width="237" alt="Captura de ecrã 2024-12-16, às 21 41 46" src="https://github.com/user-attachments/assets/dbf78a45-af40-46d4-89e7-bd2fc3f6cc8a" />

**Reply by dnewman-gpsw** [reply]
"$ is acting like if was *"  - I don't expect it is.  The read metadata displays what is active, not what is stored.   

scan: !RESET!1OR
scan: $WBDV=1!N!S!2E!1OR
scan: !S!2E

The $WBDV will not survive the reboot.

**Reply by nandodiver** [reply]
On the version **1.30.70 - November 21, 2024**

i was scanning *BITR=150 and survive reboot and was implemented in all 3 presets
i was scanning $WBLK=1 and survive to reboot and was implemented only in the scanned preset 
i was scanning $WBDV=1 and survive to reboot and was implemented only in the scanned preset

**Reply by nandodiver** [reply]
just made another !RESET and scan only one preset with $WBDV=1 and all 3 preset got metadata WBDV =1

**Reply by nandodiver** [reply]
Yes, just confirmed the $WBDV=1 did not survive to the reboot

**Reply by nandodiver** [reply]
On version **1.30.70 - November 21, 2024**
We can have specific qrcode to a specific preset and the metadata would be =0 to other presets

On version **1.30.72 - December 13, 2024**
One specific qrcode affect other presets, the metadata would be =1 to all presets

PS: See the second reply on this thread about FW 1.30.70 
the metadata show WBLK=0 and WBDV=1

**Reply by dnewman-gpsw** [reply]
I don't understand last posts.  All is working as planned in my last tests (other then all caps bug.)

**Reply by nandodiver** [reply]
ive reinstall **1.30.70 - November 21, 2024** made a !RESET, !PRESET, !FRESET and the behaviour is the same has **1.30.72 - December 13, 2024**

$ dont survive to reboot and scanned qrcode on specific preset, the metadata WBLK=1 and WBDV =1 will be present in all presets..

**Reply by dnewman-gpsw** [reply]
So it is working correctly, both versions.

**Reply by nandodiver** [reply]
Now Im testing the 1.20.70 - October 25, 2024
I was at red sea for 1 week in November filming with no issues.

**Reply by dnewman-gpsw** [reply]
Why are you going backwards, if we have confirmed that isn't anything wrong?

**Reply by nandodiver** [reply]
Im sure i had this in previous FW 

preset 1 - WBLK=1  BITR= 150 
preset 2 - WBLK=0 BITR= 150 

and i would reboot the camera and the WBLK remain =1 

My idea is that scan is not required, i just need to change presets underwater

**Reply by dnewman-gpsw** [reply]
This works in the current firmware, so use that.

*BITR=150  <- this should be global

$PRES="0,WBLK"  <- create the preset is called WBLK, only that preset will have it enabled.

**Reply by nandodiver** [reply]
this will work also ? 

$PRES="0,WBDV"  and create the preset called WBDV

**Reply by dnewman-gpsw** [reply]
Yes.

Add $PRES="0,WBDV WBLK" will do both.

**Comment by JiDey0007** [comment]
It doesn't work very well

Here I am testing and the video is almost irrecoverable on the surface, in the depths it is suitable, but finally not the expected effect, a red filter already does that

The goal was to preserve a natural image on the surface and then add red in depth

The image starts normal and after 6-8 seconds it turns red like that (at 10m depth it's already better)

For a moment I take the camera out of the water, very blue sky, the camera removes the red and the image becomes ok again (after 4 seconds)

Then again in the surface water much too red

![Capture d’écran 2024-12-27 à 09 27 00](https://github.com/user-attachments/assets/a2e4cfc7-fbcf-4d8c-a83b-1d2b40d2257b)


![Goprored](https://github.com/user-attachments/assets/3a22d828-8fe4-440a-9bde-11870fb7ea1f)

**Comment by JiDey0007** [comment]
2 or 4x gain this is not what was planned, it was planned to compensate according to the variations of the blue

See my image on the surface, it's catastrophic

**Reply by nandodiver** [reply]
> 2 or 4x gain this is not what was planned, it was planned to compensate according to the variations of the blue
> 
> See my image on the surface, it's catastrophic

WBDV is doing what is supposed to do
GAIN 4X, sky is blue, water is blue.

Did you use Lock WB ? 

Look at my last youtube videos on the link above.

**Reply by nandodiver** [reply]
@JiDey0007 i recommend you setup permanent Gopro Histogram (on display)  and use preset WBDV + WBLK 

This way you can adjust the power of your dive lights, without clipping the red using histogram.

the effect of using WBDV + WBLK is to avoid the use a external red filter, in my case is perfect because i use AOI UWL-03 wide wet lens

**Reply by JiDey0007** [reply]
Precisely the sky is blue, why does the Gopro manage to film it properly and not in the water then

No, I didn't use a WB lock, I didn't see that it was written necessary?

We don't have the same problems, you dive and stay down, your depth varies quietly then but I go back and forth over 20 m in less than 2 minutes, the interest of red management ..

I'm reading your posts

**Reply by JiDey0007** [reply]
> @JiDey0007 i recommend you setup permanent Gopro Histogram (on display) and use preset WBDV + WBLK
> 
> This way you can adjust the power of your dive lights, without clipping the red using histogram.
> 
> the effect of using WBDV + WBLK is to avoid the use a external red filter, in my case is perfect because i use AOI UWL-03 wide wet lens

You didn't understand me, I do spearfishing, no light and no diving for more than 2 minutes (between 15 and 25 m)

It's especially on the surface that red does anything

**Reply by nandodiver** [reply]
if you doing snorkel (20m depth to surface in 2 minutes) theres a specific filter for snorkel.

Or  use WBDV with 1x gain + WBLK, start the recording half way the descent, it will lock the WB after 1 second

**Reply by nandodiver** [reply]
try different values for example $WBDV=2.2,1.0

I believe 2.2 is the RBG value and 1.0 is the amount of gain.

**Reply by JiDey0007** [reply]
> if you doing snorkel (20m depth to surface in 2 minutes) theres a specific filter for snorkel.

Are you talking about a firmware or physical filter?

Thank you for your advice

**Reply by nandodiver** [reply]
theres a specific filter (physical) for snorkel

**Comment by nandodiver** [comment]
The WBDV acts like a red filter, 4x gain is too much, maybe try 2x gain 

heres my findings 
https:// waterpixels.net/forums/topic/1848-gopro13-gopro-labs-firmware-testing-underwater/#findComment-12736

**Reply by dnewman-gpsw** [reply]
This is under your control.  $WBDV=2.2,1.0 would do nothing, gain of 1.0 is means multiple by 1. I've only use some dives without lights for my experimental data, using auto white-balance. If you want less gain try $WBDV=2.33,2.0  as a 2.2 might too low and falsely correct for a blue sky.

**Reply by JiDey0007** [reply]
The example of the sky is just to say that the video is red in the water and by taking the camera out of the water a few seconds towards the sky it corrected to do a correct job

When I put the camera back in the water it became much too red again

I don't understand why it's red on the surface like that when it had to be progressive depending on the evolution of red / blue

Just not at the point?

**Reply by dnewman-gpsw** [reply]
If you share a camera original clip (even the LRV works), and I can look at the metadata to see what is going on.

**Reply by JiDey0007** [reply]
Yes, I want to be in private

**Comment by JiDey0007** [comment]
@dnewman-gpswAre you okay? We can go on WhatsApp please that I send you the private rush

**Reply by nandodiver** [reply]
you need to share original video sample or metadata

**Reply by dnewman-gpsw** [reply]
I thought you didn't want to do it. I'm not going to go chase you down.

**Reply by JiDey0007** [reply]
I want but in private
I'm the one who wouldn't force it, but only if you give me a private WhatsApp channel for example

I wrote this 2 times above 😉

**Reply by dnewman-gpsw** [reply]
"I want to be in private". Sound more like no to me, as requires me to chase you down, work not related not related to making Labs better, my only interest. If you emailed me with a download link that's fine.

**Comment by JiDey0007** [comment]
I am French, the translator may be approximate

There is material in development in the video, so if I send it it's In privately

You want to analyze why it doesn't work ok, but I don't post on the forum, it's simple, if it was no I would have said no

I will send you an email for development interest

**Comment by xiemeilong** [comment]
If I set the time for multiple GoPros using a USB serial port, will the latency be lower than with GPS time sync?

**Reply by dnewman-gpsw** [reply]
No idea. You have just volunteered to the experiment.
